src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static add(animal) {
+    Animal.alive.push(animal);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.add(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && herbivore.hidden === false) {
```

Comment:
it'd be easier to check `hidden` state with plain logical NO
```suggestion
    if (herbivore instanceof Herbivore && !herbivore.hidden) {
```